.gitlab-ci.yml

```diff
@@ -27,23 +28,45 @@ variables:
     helm --tiller-namespace ${KUBE_NAMESPACE} status "${DEPLOYMENT_NAME}"
 
 build:
+  # cache:
+  #   key: "$CI_BUILD_NAME"
+  #   paths:
+  #     - .stack-work/
+  #     - .stack-root/
   stage: build
   script:
     # Clear *_TOKEN variables during code build so that compile-time code can't access them
-    - apt-get update && apt-get install -y --no-install-recommends libpq-dev git rsync
-    - CI_BUILD_TOKEN="" KUBE_TOKEN="" PROD_KUBE_TOKEN="" PROD_DOCKER_PASSWORD="" etc/scripts/stage_docker.sh --install-ghc --test
-    - docker build -t "${DEPLOYMENT_IMAGE}" etc/docker
-    - docker login -u gitlab-ci-token -p "${CI_BUILD_TOKEN}" "${CI_REGISTRY}"
-    - docker push "${DEPLOYMENT_IMAGE}"
-    - |
-      if [[ "$CI_BUILD_REF_NAME" == "master" ]]; then
-        docker tag "${DEPLOYMENT_IMAGE}" "${CI_REGISTRY_IMAGE}:latest"
-        docker push "${CI_REGISTRY_IMAGE}:latest"
-      fi
-      if [[ "$CI_BUILD_REF_NAME" == "ci-cron" ]]; then
-        docker tag "${DEPLOYMENT_IMAGE}" "${CI_REGISTRY_IMAGE}:ci-cron"
-        docker push "${CI_REGISTRY_IMAGE}:ci-cron"
-      fi
+    - cache-s3 restore stack
+    - cache-s3 restore stack work
+    - etc/scripts/stage_docker.sh --install-ghc --test
+    - cache-s3 save stack
+    - cache-s3 save stack work
+  artifacts:
+    name: "${CI_BUILD_REF_SLUG}_${CI_PIPELINE_ID}"
+    paths:
+      - etc/docker/_artifacts/
+
+deploy_prod_s3:
+  stage: deploy
+  only:
+    - master
+  environment:
+    name: production
+  script:
+    - aws s3 cp --acl=public-read etc/docker/_artifacts/curator s3://${CI_S3_BUCKET}/${CI_S3_DIRECTORY}/curator-latest-linux-amd64
+    - aws s3 cp --acl=public-read etc/docker/_artifacts/casa-curator s3://${CI_S3_BUCKET}/${CI_S3_DIRECTORY}/casa-curator-latest-linux-amd64
+    - aws s3 cp --acl=public-read etc/docker/_artifacts/curator s3://${CI_S3_BUCKET}/${CI_S3_DIRECTORY}/curator-${CI_BUILD_TAG}-linux-amd64
```

Comment:
@JoseD92 What is the `CI_BUILD_TAG` evaluating to here? I couldn't find this in the Gitlab docs. 

We'll want this to be the git commit hash of the commit it's building.

Reply:
Ah so I couldn't find `CI_BUILD_TAG` in the docs because it's been deprecated. @JoseD92 checkout my other comment down below https://github.com/commercialhaskell/curator/pull/13#discussion_r395427001